File: ide.vstudio/ALox-CS-Test-Perf/PerformanceTest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using com.aworx.lox;
using com.aworx.util;

namespace ALox_CS_Test_Perf	 {

class PerformanceTest
{
	// #################################################################################################
	// static entrance (Main)
	// #################################################################################################
	static void Main( string[] args )
	{
		PerformanceTest test= new PerformanceTest();
		test.test();
	}

	#if ALOX_DEBUG || ALOX_REL_LOG

	// #################################################################################################
	// Fields
	// #################################################################################################
	ConsoleLogger cl;
	MemoryLogger  ml;

	#endif

	// #################################################################################################
	// Test functions
	// #################################################################################################
	
	void test()
	{
		#if ALOX_DEBUG || ALOX_REL_LOG
			cl=	new ConsoleLogger( "Console" ) { TabAfterSourceInfo = 60 };
			cl.EnableAppConsole=		true;
			cl.EnableVSDebugConsole=	false;
			ml= new MemoryLogger( "Memory");

			Log.AddLogger( cl, Log.DomainLevel.WarningsAndErrors );
			Log.AddLogger( ml, Log.DomainLevel.Off );
		#endif

		Log.RegDomain( "CON", Log.Scope.SourceFile );
		Log.RegDomain( "MEM", Log.Scope.None  );
		Log.RegDomain( "BOTH",Log.Scope.None );

		Log.SetDomain( "CON", Log.DomainLevel.All, true, "Console" );
		Log.SetDomain( "MEM", Log.DomainLevel.Off, true, "Console" );
		Log.SetDomain( "MEM", Log.DomainLevel.All, true, "Memory" );
		Log.SetDomain( "CON", Log.DomainLevel.Off, true, "Memory" );
		Log.SetDomain( "BOTH",Log.DomainLevel.All, true  );


		Log.Info( "Logging simple info lines" );

		MString	msgBuf=		new MString( );
		long	fastest=	long.MaxValue;
		for ( int i= 0 ; i < 200 ; i++ )
		{
			#if ALOX_DEBUG || ALOX_REL_LOG
				ml.Buffer.Clear();
			#endif

			long t= Ticker.Now();
				simpleInfoLines( 1000 );
			t= Ticker.Now() - t;

			if ( fastest > t )
				fastest= t;

			Log.Line( Log.Level.Info, msgBuf.Clear().Append( "Pass " ).Append( i, 3).Append( " finished") );
		}

		Log.Line( Log.Level.Info, msgBuf.Clear().Append( "Fastest " ).Append( (int) Ticker.ToMillis( fastest ), 0).Append( " millis (").Append( (int) fastest ).Append( " ticks) per 1000 logs.") );
	}

	void simpleInfoLines( int qty )
	{
		for ( int i= 0 ; i < qty ; i++ )
			Log.Info( "MEM", "Test Line" );
	}


} // class
}

File: src/java.othertests.aworx.util/perftests/com/aworx/lox/PerformanceTest.java
package perftests.com.aworx.lox;

import com.aworx.util.*;
import com.aworx.lox.*;
import com.aworx.lox.loggers.ConsoleLogger;
import com.aworx.lox.loggers.MemoryLogger;

public class PerformanceTest
{


	// #################################################################################################
	// static entrance (Main)
	// #################################################################################################
	/**
	 * @param args
	 */
	public static void main(String[] args)
	{
		PerformanceTest test= new PerformanceTest();
		test.test();
	}

	// #################################################################################################
	// Fields
	// #################################################################################################
	ConsoleLogger cl;
	MemoryLogger  ml;

	// #################################################################################################
	// Test functions
	// #################################################################################################
	
	void test()
	{
		cl=	new ConsoleLogger( "Console" ); 
		ml= new MemoryLogger( "Memory");

		Log.addLogger( cl, Log.DomainLevel.WARNINGS_AND_ERRORS );
		Log.addLogger( ml, Log.DomainLevel.OFF );

		Log.regDomain( "CON", Log.Scope.METHOD );
		Log.regDomain( "MEM", Log.Scope.NONE  );
		Log.regDomain( "BOTH",Log.Scope.NONE );

		Log.setDomain( "CON", Log.DomainLevel.ALL, true, "Console" );
		Log.setDomain( "MEM", Log.DomainLevel.OFF, true, "Console" );
		Log.setDomain( "MEM", Log.DomainLevel.ALL, true, "Memory" );
		Log.setDomain( "CON", Log.DomainLevel.OFF, true, "Memory" );
		Log.setDomain( "BOTH",Log.DomainLevel.ALL, true  );

		//Log.LogConfig( "CON", Log.Level.Info, "Log Config follows..." );
		

		Log.info( "Logging simple info lines" );
		MString msgBuf=		new MString( );
		long	fastest= 	Long.MAX_VALUE;
		for ( int i= 0 ; i < 200 ; i++ )
		{
			ml.buffer.clear();
			long t= Ticker.now();
				simpleInfoLines( 1000 );
			t= Ticker.now() - t;
			if ( fastest > t )
 				fastest= t;

			Log.line( Log.Level.INFO, msgBuf.clear().append( "Pass " ).append( i, 3).append( " finished") );
		}

		Log.line( Log.Level.INFO, msgBuf.clear().append( "Fastest " ).append( (int) Ticker.toMillis( fastest ), 0).append( " millis (").append( (int) fastest ).append( " ticks)  per 1000 logs.") );
	}

	static void simpleInfoLines( int qty )
	{
		for ( int i= 0 ; i < qty ; i++ )
			Log.info( "MEM", "Test Line" );
	}

}
